src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import core.basesyntax.models.*;
+import java.util.Random;
+
+class FigureSupplier {
+    private final ColorSupplier colorSupplier;
+    private final Random random;
+
+    public FigureSupplier() {
+        this.colorSupplier = new ColorSupplier();
+        this.random = new Random();
+    }
+
+    public Figure getRandomFigure() {
+        int type = random.nextInt(5);
+        String color = colorSupplier.getRandomColor();
+
+        return switch (type) {
+            case 0 -> new Square(color, random.nextDouble() * 10 + 1);

```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables for the size parameter and pass them on for better code readability.